pkg/cmd/sync/sync_integration_test.go

```diff
@@ -0,0 +1,86 @@
+// +build integration
+
+package sync_test
+
+import (
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/jenkins-x/jx/pkg/cmd/clients"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/sync"
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/ksync"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var sleepbeforecheck time.Duration = 30
+
+func TestKsyncCmd(t *testing.T) {
+	origJXHome, testJXHome, err := testhelpers.CreateTestJxHomeDir()
+	assert.NoError(t, err, "failed to create a test JX Home directory")
+
+	defer func() {
+		// Delete the pods in the test kubernetes cluster
+		kcli, err := ksync.NewCLI()
+		if err != nil {
+			log.Logger().Warnf("unable to create ksync cli client. Error: %s", err)
+		}
+
+		_, err = kcli.Clean()
+		if err != nil {
+			log.Logger().Warnf("unable to remove ksync daemonset. Error: %s", err)
+		}
+		log.Logger().Info("Removed ksync pods from cluster")
+
+		// Delete text directories
+		err = testhelpers.CleanupTestJxHomeDir(origJXHome, testJXHome)
+		if err != nil {
+			log.Logger().Warnf("unable to remove temporary directory %s: %s", testJXHome, err)
+		}
+		log.Logger().Info("Removed temp directory")
+	}()
+
+	commonOpts := opts.NewCommonOptionsWithTerm(clients.NewFactory(), os.Stdin, os.Stdout, os.Stderr)
+	assert.NoError(t, err, "error creating kubeclient")
+	cmd := sync.NewCmdSync(commonOpts)
+
+	executeCh := make(chan error, 1)
+	checkCh := make(chan error, 1)
+
+	// Run ksync in it's own goroutine and pass the error message into a channel
+	go func(cmd *cobra.Command) {
+		err := cmd.Execute()
+		executeCh <- err
+	}(cmd)
+
+	// Check for ksync daemonset in another go routine and pass the error to a channel
+	go func(opts *opts.CommonOptions) {
+		// Wait for at least one pod to be ready
+		time.Sleep(sleepbeforecheck * time.Second)
+		options := metav1.GetOptions{}
+		client, err := opts.KubeClient()
+		assert.NoError(t, err, "cannot create kube client")
+		ds, err := client.AppsV1().DaemonSets("kube-system").Get("ksync", options)
+		assert.NoError(t, err, "Cannot get ksync pods in kube-system namespace")
+		ready := ds.Status.NumberReady
+		// If none of the pods are ready
+		if ready == 0 {
+			checkCh <- errors.New("zero pods ready")
+		}
+		checkCh <- nil
+	}(commonOpts)
+
+	select {
+	case err := <-executeCh:
+		assert.NoError(t, err, "unable to execute ksync commands")
+	case err := <-checkCh:
+		assert.NoError(t, err, "ksync pods not ready")
```

Comment:
Maybe also here add a time based select, so in case something goes wrong the test will eventually time out.